src/natcap/invest/delineateit/delineateit.py

```diff
@@ -601,6 +630,193 @@ def snap_points_to_nearest_stream(points_vector_path, stream_raster_path_band,
     points_vector = None
 
 
+def detect_pour_points(flow_dir_raster_path_band, target_vector_path):
+    """
+    Create a pour point vector from D8 flow direction raster.
+
+    A pour point is the center point of a pixel which:
+        - flows off of the raster, or
+        - flows into a nodata pixel
+
+    Args:
+        flow_dir_raster_path_band (tuple): tuple of (raster path, band index)
+            indicating the flow direction raster to use. Pixel values are D8
+            values [0 - 7] in this order:
+                321
+                4x0
+                567
+        target_vector_path (string): path to save pour point vector to.
+
+    Returns:
+        None
+    """
+    raster_info = pygeoprocessing.get_raster_info(flow_dir_raster_path_band[0])
+    pour_point_set = _find_raster_pour_points(flow_dir_raster_path_band, 
+                                              raster_info)
```

Comment:
I was conflicted about what to do here because the `raster_info` is needed in both `detect_pour_points` and `_find_raster_pour_points`. I originally had `get_raster_info` called in both functions, then I changed it to be passed into the function so that it's only called once. I'm not really happy with either solution...